eureka-client/src/main/java/com/netflix/appinfo/CloudInstanceConfig.java

```diff
@@ -50,7 +50,7 @@
     };
 
     private final AmazonInfoConfig amazonInfoConfig;
-    private final RefreshableAmazonInfoProvider amazonInfoHolder;
+    /* visible for testing */ RefreshableAmazonInfoProvider amazonInfoHolder;
```

Comment:
I would appreciate this be kept final to better ensure it is always correctly instantiated. An alternative for your test can be to add a new constructor that can be used to set `RefreshableAmazonInfoProvider`, with test visibility?

Reply:
Fixed.